### DISCUSSION TITLE
How to leverage pre-trained DNN for CEBRA?

### DISCUSSION THREAD

**Discussion by jungwoooseo** [AdaptiveMotorControlLab/CEBRA#94] [Q&A] [answered]
Hi authors, first of all, thank you for your amazing work. 
We are planning to use CEBRA for our research and are considering leveraging a pre-trained encoder for performance improvement. After reading the CEBRA usage notes, we realized that can define a custom model architecture to train, but it seemed unclear whether we can import and use a pre-trained model. So, I'm asking directly: Can we load and use a pre-trained encoder in CEBRA and fine-tune it on the CEBRA framework?

**Comment by stes** [comment, marked as answer]
Hi @jungwoooseo , yes that is possible. You need to define a model architecture for your pretrained encoder (see cebra/models/model for examples on how to register).

Preload your weights during initialisation, and then use the model either through the sklearn/torch API. In sklearn you'd pass the architecture Name you used to register the encoder.

Let me know if you have additional questions!

**Reply by jungwoooseo** [reply]
Thanks :) I'll try.
